geps/gep-1619/index.md

```diff
@@ -459,52 +438,46 @@ route in any given implementation.
-without defaults or overrides, applied to the targeted service, HTTPRoute, or GRPCRoute.
+In order to apply session persistence configuration to a backend, we will implement it as a metaresource policy, as
+outlined in [Policy Attachment](/reference/policy-attachment/). The new metaresource is named `LoadBalancerPolicy` and
+is responsible for configuring load balancing-related configuration for traffic intended for a backend after routing has
+occurred. It is defined as a [Direct Policy Attachment](/geps/gep-713/#direct-policy-attachment) without defaults or
+overrides, applied to the targeted backend.
 
-Attaching the `SessionPersistencePolicy` metaresource to a service will be a core support level feature, while attaching
-it to a route will be considered extended. Implementations must support services for conformance, while routes are
-considered optional. This distinction arises because most existing implementations primarily support attaching to a
-service, while attaching to a route is less common and involves greater complexities (see [API Attachment Points](#api-attachment-points)
-for more details).
+Instead of utilizing a specific, session persistence-only policy object, we introduce a more generic API object named
+`LoadBalancerPolicy`. This design provides tighter coupling with other load balancing configuration which helps reduce
+CRD proliferation. For instance, `LoadBalancerPolicy` could be augmented to add configuration for selecting a load
+balancing algorithm for traffic to the backends, as desired in issue [#1778](https://github.com/kubernetes-sigs/gateway-api/issues/1778).
+`LoadBalancerPolicy` could also be later expanded to contain [session affinity](#the-relationship-of-session-persistence-and-session-affinity)
+configuration. This would provide a convenient grouping of the two related APIs within the same policy object.
+Additionally, other future enhancements to the API may include the addition of timeouts, connection draining, and
+logging within `LoadBalancerPolicy`.
 
-### API Attachment Points
+### API Attachment Point
 
-The `SessionPersistencePolicy` metaresource can target a service, route, and a route rule section (e.g. [HTTPRouteRule](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule)).
-The core functionality of attaching it to any of these is generally identical. The distinction lies in where the
-configuration gets propagated: when attached to a route or a route rule, it can define session persistence for multiple
-services, whereas attaching it to a service defines it for a single service. Enabling session persistence doesn't
-require configuration on the route, route rule, and service; configuring it on any one is sufficient.
+Attaching the `LoadBalancerPolicy` metaresource to a backend will be a core support level feature, while attaching
+it to a route or a route rule section may be addressed in a future update of this GEP. A backend can be a Service,
```

Comment:
I'd echo @ginayeh's thoughts here. Attaching session persistence config via Route rule helps avoid many of the edge cases we deal with when it comes to backend attachment. For example, when traffic splitting it would be much easier/cleaner to have config that applied for the entire route rule instead of separately for each backend.

Reply:
I am not against having route attachment ( in addition to Service) - besides persistent sessions there are few other things that could be set for route. 

I don't think we have any requirement to use the exact same CRD in both places - I think it is very reasonable to have a backend policy that specifies load balancing and related functions - with different kinds of persistence impacting load balancing. And we can have a slightly different resource per route:  some of the concerns with the route - for example hard to define behaviors if the cookie name or other settings are set on multiple routes - can be resolved by having a simpler per-route override, for example just ability to enable/disable persistence - while keeping the details on the load balancing policies. 

Envoy has the ability to do per-route - others may not, so the per route could be an advanced feature. I expect it may take a bit of time for implementations to identify corner cases ( routes in different namespaces setting different things) and other proxies deciding if they want to support per route, but if we start with a minimal per-route API, just like we reduced the per-backend API - it may be quite useful.

Reply:
To clarify: I like the rename to LoadBalancerPolicy and ability to have multiple things impacting endpoint selection here, as a backend policy. 

For per-route - it seems useful to also have a policy with related overrides, but I don't think we should force LoadBalancerPolicy to be used for both.

Reply:
Updated to add back route support as an inline route-rule. Curious to hear thoughts. I kept the session persistence config independent like @costinm suggested.